### ISSUE TITLE
add chromosome and position to basic gene info JSON schema and files

### ISSUE THREAD

**Issue by sierra-moxon** [alliance-genome/agr_schemas#14]
Last week I discussed genome info with the BGI-WG.  We asked today in the technical call, if that can come from genome features working group as implied by BGI-WG.   The GF-WG would rather it come from the basic gene info.

We need to see if we can get this into 0.3

**Comment by khowe** [comment]
Smells like a new genomic_location schema:
{
   assembly => "GCA_XXXXX.1", 
   sequence => "III",
   start  => 100000,
   end   => 1100000
}

(note that some species have genes located on unlocalised and/or unmapped sequence (i.e. no associated with a chromosome yet)

**Comment by christabone** [comment]
Would anyone have time to sit in on their call starting in 20 minutes? I'll do my best to be there but I might be working on a few things locally.

**Comment by christabone** [comment]
@khowe Should we add it to the current schema or start a new file?

**Comment by khowe** [comment]
@christabone I guess a trend has been set by the previous work for doing it in a separate file. 

Another thought: need to make it non-compulsory (uncloned genes will not yet have a genomic location; although I've excluded such genes for the WB export)

**Comment by sierra-moxon** [comment]
The separate file would be nice if we want to give non-gene features this same genomic_location treatment.

**Comment by nathandunn** [comment]
@sierra-moxon Was just typing myself ;) 

FYI @scottcain , @pjhale:

tl;dr To generate a URL for JBrowse we need the base URL for the organism (or taxonId) / assemblage, the chromosome (whatever is used internally), and the location (start/end).   

In lieu of this, you could use the URL's directly from the schema, but not sure if that is what their purpose is:

https://github.com/alliance-genome/agr_schemas/blob/development/basicGeneInformation.json#L30-L33

https://github.com/alliance-genome/agr_schemas/blob/development/basicGeneInformation.json#L49-L56

Here is a sample URL:

https://genome.monarchinitiative.org/apollo/Honeybee/jbrowse/index.html?loc=Group1.1%3A724015..726599&tracklist=0&nav=0&tracks=Official%20Gene%20Set%20v3.2

The relative parts:
- host URL / base URL
- Organism / taxonId
- Sequence
- Start / End
- Track (just the name of the one - could be made to be consistent, though)

All of the other options are just dictate the UI and may not even be used.

**Comment by christabone** [comment]
non-BlueJeans link for meeting: https://lbnl.zoom.us/j/4161956725

**Comment by khowe** [comment]
@sierra-moxon good point about reusing the schema for other types of feature. 

Definite thumbs down from me though for including the sequence itself in the JSON (what sequence? gene locus? transcripts? spliced_transcripts? etc etc. Can of worms)

**Comment by monicacecilia** [comment]
Thanks for pointing this out @sierra-moxon. Since coordinates are the driving data for creating the image for a locus and all transcripts in that location, we can have more than one image per gene page, one for each of its locations.

**Comment by scottcain** [comment]
@monicacecilia and @sierra-moxon we can have more than one image per page, assuming whoever is building the page knows to look for more than one set of coordinates and behaves accordingly, that is.  In any event, the schema needs to be able to capture that sort of "array of location objects" (or whatever you'd choose to call it) information.

**Comment by nathandunn** [comment]
Also noted by @selewis  it would be a good idea to indicate the assembly if available.   @scottcain  said that he is going to put all the data on a central server so we should have a central URL that would roughly correspond to this:

<hard-coded server url>/<taxonId>/?loc=<sequence>:<start>..<end>

**Comment by sierra-moxon** [comment]
Ok - and we also have features with only assembly x and others with only assembly y.   Said another way, the entire set of features from ZFIN won't be on the same assembly.  Still ok?

**Comment by nathandunn** [comment]
My assumption would be yes.    We can always filter to show one assembly or another (or both) if need be.

**Comment by scottcain** [comment]
@sierra-moxon I don't see why that would be a problem.  I would do the same thing that zfin is doing with your gbrowse--two data sources, one for each assembly.
